lib/discourse_reactions/topic_view_serializer_extension.rb

```diff
@@ -29,8 +35,7 @@ def posts
         )
 
       posts.each do |post|
-        post.post_actions_with_reaction_users =
-          post_actions_with_reaction_users.select { |post_action| post_action.post_id == post.id }
```

Comment:
This was looping through the array once per post which is inefficient. Instead, we reduce the array into a hash keyed by the `post_action.post_id` instead so that we can easily access it later.